src/bptl/work_units/valid_sign/tasks.py

```diff
@@ -0,0 +1,283 @@
+
+            self._validsign_client = services_by_alias[alias].build_client()
+        return self._validsign_client
+
+    def perform(self) -> dict:
+        raise NotImplementedError
+
+
+@register
+class CreateValidSignPackageTask(ValidSignTask):
+    """Create a ValidSign package with signers and documents and send a signing request to the signers.
+
+    **Required process variables**
+
+    * ``documents``: List of strings. List of API URLs where the documents to be signed can be retrieved.
+        The API must comply with the Documenten API 1.0.x (https://vng-realisatie.github.io/gemma-zaken/standaard/documenten/index).
+
+    * ``signers``: JSON list with signers information. For ValidSign, the first name, the last name and the
+        email address of each signer are required. Example ``signers``:
+
+            .. code-block:: json
+
+                [{
+                    "email": "example.signer@example.com",
+                    "firstName": "ExampleFirstName",
+                    "lastName": "ExampleLastName"
+                },
+                {
+                    "email": "another.signer@example.com",
+                    "firstName": "AnotherFirstName",
+                    "lastName": "AnotherLastName"
+                }]
+
+    * ``packageName``: string. Name of the ValidSign package that contains the documents to sign and the signers.
+        This name appears in the notification-email that is sent to the signers.
+
+    **Sets the process variables**
+
+    * ``packageId``: string. ID of the ValidSign package created by the task.
+    """
+
+    _documents_client = None
+
+    def get_documents_client(self, topic_name: str) -> ZGWClient:
+        if self._documents_client is None:
+            default_services = TaskMapping.objects.get(
+                topic_name=topic_name
+            ).defaultservice_set.select_related("service")
+            services_by_alias = {svc.alias: svc.service for svc in default_services}
+
+            alias = "DocumentenAPI"
+            if alias not in services_by_alias:
+                raise RuntimeError(f"Service alias '{alias}' not found.")
+
+            self._documents_client = services_by_alias[alias].build_client()
+        return self._documents_client
+
+    def format_signers(self, signers: List[dict]) -> List[dict]:
+        """Format the signer information into an array of JSON objects as needed by ValidSign."""
+
+        return [{"type": "SIGNER", "signers": [signer]} for signer in signers]
+
+    def _get_documents_from_api(self) -> List[Tuple[str, bytes]]:
+        """Retrieve the documents and their content from the Documenten API."""
+
+        logger.debug("Retrieving documents from Documenten API")
+
+        variables = self.task.get_variables()
+        document_urls = check_variable(variables, "documents")
+        document_client = self.get_documents_client(topic_name="CreateValidSignPackage")
```

Comment:
hard-coded topic name is not going to work - the topic name is dynamic and specified when you create `TaskMapping` objects. You can retrieve it from `self.task` though.

Reply:
Updated!